src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const { type, keysToRemove, extraData } of actions) {
+    switch (type) {
+      case 'addProperties':
+        addProps(currentState, extraData);
+        break;
+      case 'removeProperties':
+        removeProps(currentState, keysToRemove);
+        break;
+      default:
+        currentState = {};
```

Comment:
While you have used a `switch` statement, which is good for handling different types of actions, you should include a `clear` case to handle the action type 'clear'. Additionally, the default case should not clear the state; it should either do nothing or throw an error to handle unexpected action types.